pkg/clients/cloudantdatabase/cloudantdatabase.go

```diff
@@ -0,0 +1,130 @@
+func LateInitializeSpec(spec *v1alpha1.CloudantDatabaseParameters, in *cv1.DatabaseInformation) error { // nolint:gocyclo
+
+	if spec.Partitioned == nil {
+		spec.Partitioned = in.Props.Partitioned
+	}
+
+	if spec.Q == nil {
+		spec.Q = in.Cluster.Q
+	}
+
+	return nil
+}
+
+// GenerateCreateCloudantDatabaseOptions produces PutDatabaseOptions object from CloudantDatabaseParameters object.
+func GenerateCreateCloudantDatabaseOptions(in v1alpha1.CloudantDatabaseParameters, o *cv1.PutDatabaseOptions) error {
+	o.Db = reference.ToPtrValue(in.Db)
+	o.Partitioned = in.Partitioned
+	o.Q = in.Q
+	return nil
+}
+
+// GenerateObservation produces CloudantDatabaseObservation object from *cv1.DatabaseInformation object.
+func GenerateObservation(in *cv1.DatabaseInformation) (v1alpha1.CloudantDatabaseObservation, error) {
+	o := v1alpha1.CloudantDatabaseObservation{
+		Cluster:            Generatev1alpha1DatabaseInformationCluster(in.Cluster),
+		CommittedUpdateSeq: reference.FromPtrValue(in.CommittedUpdateSeq),
+		CompactRunning:     ibmc.BoolValue(in.CompactRunning),
+		CompactedSeq:       reference.FromPtrValue(in.CompactedSeq),
+		DiskFormatVersion:  ibmc.Int64Value(in.DiskFormatVersion),
+		DocCount:           ibmc.Int64Value(in.DocCount),
+		DocDelCount:        ibmc.Int64Value(in.DocDelCount),
+		Engine:             reference.FromPtrValue(in.Engine),
+		// not including props for now because I don't think it should be in the API ??
+		// Props: in.Props,
+		Sizes:     Generatev1alpha1ContentInformationSizes(in.Sizes),
+		UpdateSeq: reference.FromPtrValue(in.UpdateSeq),
+		UUID:      reference.FromPtrValue(in.UUID),
+	}
+	return o, nil
+}
+
+// Generatev1alpha1DatabaseInformationCluster generates *v1alpha1.DatabaseinformationCluster from *cv1.DatabaseInformationCluster
+func Generatev1alpha1DatabaseInformationCluster(in *cv1.DatabaseInformationCluster) *v1alpha1.DatabaseInformationCluster {
+	if in == nil {
+		return nil
+	}
+	o := &v1alpha1.DatabaseInformationCluster{
+		N: ibmc.Int64Value(in.N),
+		R: ibmc.Int64Value(in.R),
+		W: ibmc.Int64Value(in.W),
+	}
+	return o
+}
+
+// Generatev1alpha1ContentInformationSizes generates *v1alpha1.ContentInformationSizes from *cv1.ContentInformationSizes
+func Generatev1alpha1ContentInformationSizes(in *cv1.ContentInformationSizes) *v1alpha1.ContentInformationSizes {
+	if in == nil {
+		return nil
+	}
+	o := &v1alpha1.ContentInformationSizes{
+		Active:   ibmc.Int64Value(in.Active),
+		External: ibmc.Int64Value(in.External),
+		File:     ibmc.Int64Value(in.File),
+	}
+	return o
+}
+
+// IsUpToDate checks whether current state is up-to-date compared to the given set of parameters.
+func IsUpToDate(in *v1alpha1.CloudantDatabaseParameters, observed *cv1.DatabaseInformation, l logging.Logger) (bool, error) {
+	desired := in.DeepCopy()
+	actual, err := GenerateCloudantDatabaseParameters(observed)
+	if err != nil {
+		return false, err
+	}
+	// is this right ??
```

Comment:
Looks good, you might need to add fields in IgnoreFields(v1alpha1.CloudantDatabaseParameters{} based on how the unit tests will work out